[Needs review] add zoom range "min..max" options for burn + other changes

vincentsarago opened the pull request:
👋 @dnomadb 
I think having the possibility to pass a zoom range to the `burn` command could be a nice feature (at least for myself). I tried to apply the same logic at in rio-mbtiles where we pass the Min Max zoom as a `min..max` string. 

```
rio bounds my_file.tif | supermercado burn 18..21 | xt -d'-' > list_tiles.txt
```

I know this could be achieve with a bash onliner but you know I don't love bash 😄 

I also did some other change to improve code readability and help. 

cc @sgillies

dnomadb commented on supermercado/scripts/cli.py:
```diff
@@ -7,7 +10,8 @@
 
-@click.command('edges')
+
+@cli.command('edges', short_help="Return egde tiles for a stream of [<x>, <y>, <z>] tiles.")
```
`egde` --> `edge`

vincentsarago replied:
?

dnomadb replied:
> Return egde tiles

vincentsarago replied:
🤦‍♂️

dnomadb commented on supermercado/scripts/cli.py:
```diff
@@ -42,22 +45,28 @@ def union(inputtiles, parsenames):
+    minzoom = zooms[0]
+    maxzoom = zooms[0] if len(zooms) == 1 else zooms[1]
 
+    tiles = []
```
Why not:

```
tiles = (burntiles.burn(features, zoom) for zoom in range(minzoom, maxzoom + 1))
```

dnomadb commented on supermercado/scripts/cli.py:
```diff
@@ -42,22 +45,28 @@ def union(inputtiles, parsenames):
 
-cli.add_command(burn)
\ No newline at end of file
+    for t in list(itertools.chain.from_iterable(tiles)):
```
I don't think we need to make this a list before iterating over all the output tiles.

dnomadb commented on supermercado/scripts/cli.py:
```diff
@@ -42,22 +45,28 @@ def union(inputtiles, parsenames):
-@click.argument('zoom', type=int)
+@click.argument(
+    'zoom',
+    type=str
```
The downside of this is that we lose click's argument validation -- we'd need to add a bunch of checks for:
- is the string formatted correctly
- is the parsed output integers
- are they in the right order (eg what if someone wrote `10..0`)

vincentsarago replied:
Thanks @dnomadb, 
I could create a custom class for that like https://github.com/mapbox/rio-glui/blob/master/rio_glui/scripts/cli.py#L35-L52

vincentsarago replied:
> are they in the right order (eg what if someone wrote 10..0)

well this might be tricky but we can use `min()` and `max()` to make sure of the order

vincentsarago commented:
👋 @dnomadb I've updated the code and added a custom `zoom type` to make sure the input is 👌. 
I've also updated the tests but sadly they are failing due to a depreciation in numpy inside `findedges` function

```
  File /edge_finder.py", line 26, in findedges
    )), axis=2) - burn)
TypeError: numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logical_xor function instead.
```

dnomadb commented on supermercado/scripts/cli.py:
```diff
@@ -42,22 +46,43 @@ def union(inputtiles, parsenames):
-    for t in tiles:
-        click.echo(t.tolist())
+    tiles = (
+        burntiles.burn(features, zoom) for zoom in range(zoom[0], zoom[1] + 1)
```
We don't have to re-burn for every zoom; we can use the maxzoom to generate the tile-cover, then:
```python
tiles[:, :2] >>= 1
tiles[:, 2] -= 1
np.unique(tiles, axis=0)
```
This will output the parents of `tiles`. cc @pratikyadav

vincentsarago replied:
yeah makes sense 👍

sgillies replied:
Explain @dnomadb ? Does that not risk false positives, sweeping in small child tiles that are outside the features?

dnomadb replied:
@sgillies y of course!

A feature that covers a tile at zoom z will always cover said tiles at zooms z-n. False positives would be a (huge) risk going to _higher_ zooms but if we perform the tile cover at the _maxzoom_ of the provided range, we can then derive lower zooms directly from this array.

sgillies replied:
Right! Thanks, I must have had some min/max zoom confusion in my head at the end of the day.
